check/tests/pass.rs

```diff
@@ -871,3 +871,16 @@ type Bar = Test Foo
     let result = support::typecheck(text);
     assert!(result.is_ok(), "{}", result.unwrap_err());
 }
+
+#[test]
+fn dont_guess_a_record_when_the_construction_has_no_fields() {
+    let _ = ::env_logger::init();
+    let text = r#"
+type Test = { x : Int }
+type Test2 = Int
+
+{ Test2 }
+"#;
+    let result = support::typecheck(text);
+    assert!(result.is_ok(), "{}", result.unwrap_err());
+}
```

Comment:
The `find_record` function takes `&[Symbol]` and looks for a record which contains all of those fields so if the slice is empty it will match on any record (empty set is a subset of all sets).